exonum/src/blockchain/schema.rs

```diff
@@ -378,7 +383,12 @@ where
 /// an error may occur. Since Exonum services may call each other's methods, `CallSite` is
 /// richer than `CallInBlock`.
 ///
+/// One example of difference between these types is [`CallType::Constructor`]: since services
+/// are constructed outside of the block processing routine, this kind of errors cannot be
```

Comment:
I don't quite understand what the "block processing routine" is supposed to mean. Services *are* constructed as a result of usual block calls to the supervisor service, aren't they? (Well, builtin services being an exception.) The difference is that service constructors are not topmost calls.

Reply:
What I meant here is that service creation happens outside of "interacting points" within block (`before/after_transactions` hook, transaction execution). 

From the point of users view, services are constructed after `after_transactions` hooks, and there is no such a point of interaction for them, it's totally under the hood.

Do you have a suggestion how to phrase it with less ambiguity?

Reply:
> services are constructed after `after_transactions` hooks

Ah, I understand now; we've got a terminology issue. I've understood "service creation" to be equivalent to `Runtime::initiate_starting_service` calls, and these calls happen *within* "interacting points" of a block. Instantiating service objects indeed happens both in these calls and during block commitment.

Reply:
Yeah, but invocation of `Runtime::initiate_starting_service` is hidden from users, so I decided that it's better to not disclose it.